File: backend/app/schemas/log.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from app.schemas.watchlist import CamelModel

LogLevel = Literal["OK", "INFO", "WARN", "ERROR"]


class LogEntryOut(CamelModel):
    id: int
    level: LogLevel
    source: str
    message: str
    detail: str | None = None
    created_at: datetime

File: frontend/src/types/log.ts
export type LogLevel = 'OK' | 'INFO' | 'WARN' | 'ERROR'

export const LOG_LEVELS: LogLevel[] = ['OK', 'INFO', 'WARN', 'ERROR']

export type LogLevelFilterValue = 'ALL' | LogLevel

export const LOG_LEVEL_FILTERS: LogLevelFilterValue[] = ['ALL', 'OK', 'INFO', 'WARN', 'ERROR']

export interface LogEntry {
  id: number
  level: LogLevel
  source: string
  message: string
  detail: string | null
  createdAt: string
}
